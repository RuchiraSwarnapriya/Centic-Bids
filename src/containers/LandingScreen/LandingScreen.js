import React from 'react';
import { View, StyleSheet, SafeAreaView, Image } from 'react-native';
import { LOGIN, GUESTHOME } from '../../navigation/routePaths';
import { Colors } from "../../assets/colors";
import AppLogo from "../../assets/images/logo.png";
import Button from "../../components/button/Button";


const LandingScreen = ({ navigation }) => {

    //navigate to login page
    const LoginNavigatgion = () => navigation.navigate(LOGIN);

    //navigate to guest home page
    const GuestNavigation = () => navigation.navigate(GUESTHOME);

    return (
        <SafeAreaView style={styles.container}>
            <Image source={AppLogo} style={styles.logo}></Image>
            <View>
                <Button
                    title="Login"
                    buttonStyle={styles.loginButton}
                    onPress={LoginNavigatgion}
                />
                <View style={styles.separator}></View>
                <Button
                    title="Guest"
                    buttonStyle={styles.guestButton}
                    onPress={GuestNavigation}
                />
            </View>
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'flex-end',
        paddingHorizontal: 16,
        paddingBottom: 50,
        backgroundColor: Colors.white
    },
    separator: {
        alignSelf: 'center',
        width: 250,
        marginVertical: 10,
        borderBottomColor: Colors.darkGrey,
        borderBottomWidth: StyleSheet.hairlineWidth,
    },
    logo: {
        justifyContent: 'flex-start',
        alignSelf: 'center',
        width: 300,
        height: 300,
        bottom: 100
    },
    loginButton: {
        backgroundColor: Colors.darkMegenda
    },
    guestButton: {
        backgroundColor: Colors.green
    }
});

export default LandingScreen
