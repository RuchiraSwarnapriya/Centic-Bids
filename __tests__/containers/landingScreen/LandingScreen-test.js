import React from 'react';
import renderer from 'react-test-renderer';
import LandingScreen from "../../../src/containers/landingScreen/LandingScreen";

jest.mock('@react-native-firebase/auth');

jest.mock("react-native/Libraries/EventEmitter/NativeEventEmitter");

test('renders correctly', () => {
    const tree = renderer.create(< LandingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});