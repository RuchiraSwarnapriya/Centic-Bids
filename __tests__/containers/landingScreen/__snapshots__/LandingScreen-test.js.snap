// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "flex-end",
      "paddingBottom": 50,
      "paddingHorizontal": 16,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "bottom": 100,
        "height": 300,
        "justifyContent": "flex-start",
        "width": 300,
      }
    }
  />
  <View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#841584",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderBottomColor": "#A5A5A7",
          "borderBottomWidth": 0.5,
          "marginVertical": 10,
          "width": 250,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#008000",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Guest
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
